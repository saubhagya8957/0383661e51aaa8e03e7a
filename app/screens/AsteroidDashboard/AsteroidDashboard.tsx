import React, { Component } from "react";
import { View, SafeAreaView, Text, TextInput, TouchableOpacity, Alert, ActivityIndicator } from 'react-native';
import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
import styles from './AsteroidDashboardStyle';


interface Props {
    navigation: any;
    id: string;
}

interface S {
    asteroidData: any;
    asteroidValue: any;
    asteroidDataList: any;
    showLoader: boolean;
}

interface SS {
    id: any
}

class AsteroidDashboard extends Component<Props, S, SS> {
    constructor(props: Props) {
        super(props);
        this.state = {
            asteroidValue: '',
            asteroidData: null,
            asteroidDataList: [],
            showLoader: false,
        }
    }

    componentDidMount = () => {

    }

    componentWillUnmount = () => {

    }

    onFindAsteroidData = async () => {
        this.setState({ showLoader: true });
        try {
            let asteroidResponse = await fetch(`https://api.nasa.gov/neo/rest/v1/neo/${this.state.asteroidValue}?api_key=o65fRk7BljKimvXXqaIqaloMrmgyOuV74vy29lP2`);
            let asteroidJson: any = await asteroidResponse.json();
            this.setState({ showLoader: false, asteroidData: asteroidJson }, () => {
                if(asteroidJson.status === 404) {
                    Alert.alert('Invalid asteroid ID, please check it again');
                    return;
                }
                this.props.navigation.navigate('AsteroidDataList', { asteroidData: this.state.asteroidData });
                console.log('@@@ Asteroid JSON Response ===========', `https://api.nasa.gov/neo/rest/v1/neo/${this.state.asteroidValue}?api_key=o65fRk7BljKimvXXqaIqaloMrmgyOuV74vy29lP2`, this.state.asteroidData);
            })
        } catch (error) {
            console.log('@@@ Error Asteroid ===========', error);
        }
    }

    onFindRandomAsteroidData = async () => {
        this.setState({ showLoader: true });
        try {
            let asteroidResponse = await fetch(`https://api.nasa.gov/neo/rest/v1/neo/browse?api_key=o65fRk7BljKimvXXqaIqaloMrmgyOuV74vy29lP2`);
            let asteroidJson: any = await asteroidResponse.json();
            this.setState({ showLoader: false, asteroidDataList: asteroidJson.near_earth_objects }, () => {
                if(asteroidJson.status === 404) {
                    Alert.alert('Invalid asteroid ID, please check it again');
                    return;
                }
                console.log('@@@ Asteroid Random JSON Response ===========', this.state.asteroidDataList);
                this.props.navigation.navigate('AsteroidListScreen', { asteroidDataList: this.state.asteroidDataList });
            })
        } catch (error) {
            console.log('@@@ Error Asteroid ===========', error);
        }
    }

    renderLoader = () => {
        return (
            <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', position: 'absolute' }}>
                <ActivityIndicator size="large" color={'#ffffff'} />
            </View>
        )
    }

    render() {
        const isSubmitDisabled = this.state.asteroidValue.trim().length === 0;
        return (
            <SafeAreaView style={styles.container}>
                <KeyboardAwareScrollView contentContainerStyle={{ flex: 1 }}>
                    <View style={styles.asteroidForm}>
                        <Text style={styles.asteroidHeadingText}>Search Asteroid Data</Text>
                        <TextInput
                            testID="searchTextInputID"
                            style={styles.inputStyle}
                            placeholder={'Enter Asteroid ID'}
                            placeholderTextColor={'#ffffff'}
                            value={this.state.asteroidValue}
                            onChangeText={(value) => this.setState({ asteroidValue: value })}
                            keyboardType={'numeric'}
                        />
                        <TouchableOpacity onPress={() => this.onFindAsteroidData()} disabled={isSubmitDisabled} testID="submitButtonTestID" style={[styles.submitButtonStyle, { backgroundColor: isSubmitDisabled ? '#757575' : '#00c92f'}]}>
                            <Text style={styles.submitText}>Submit</Text>
                        </TouchableOpacity>
                        <TouchableOpacity onPress={() => this.onFindRandomAsteroidData()} testID="randomButtonTestID" style={styles.randomButtonStyle}>
                            <Text style={styles.randomText}>Random Asteroid</Text>
                        </TouchableOpacity>
                    </View>
                </KeyboardAwareScrollView>
                {this.state.showLoader && this.renderLoader()}
            </SafeAreaView>
        );
    }
}

export default AsteroidDashboard;