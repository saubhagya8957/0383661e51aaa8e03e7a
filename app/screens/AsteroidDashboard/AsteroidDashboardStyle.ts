import { StyleSheet } from "react-native";


export default StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#ab8e00'
    },

    asteroidForm: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#ab8e00'
    },

    asteroidHeadingText: {
        fontSize: 30,
        textAlign: 'center',
        fontWeight: 'bold',
        color: '#e8f4ff',
        marginVertical: 50
    },

    inputStyle: {
        alignSelf: 'center',
        width: 300,
        height: 50,
        color: '#000000',
        borderWidth: 2,
        borderColor: '#000000',
        marginTop: 20,
        paddingLeft: 20
    },

    submitButtonStyle: {
        width: 250,
        height: 50,
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 15,
        backgroundColor: '#000000',
        marginTop: 20,
    },

    submitText: {
        fontSize: 20,
        fontWeight: 'bold',
        color: '#ffffff'
    },

    randomButtonStyle: {
        width: 250,
        height: 50,
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 15,
        backgroundColor: '#3000a8',
        marginTop: 20,
    },

    randomText: {
        fontSize: 20,
        fontWeight: 'bold',
        color: '#ffffff'
    },
})  