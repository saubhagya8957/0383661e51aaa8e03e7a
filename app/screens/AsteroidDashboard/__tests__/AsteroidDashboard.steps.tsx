import React from "react";
import { defineFeature, loadFeature } from 'jest-cucumber';
import { shallow, ShallowWrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AsteroidDashboard from "../AsteroidDashboard";

configure({ adapter : new Adapter() });

const screenProps = {
    navigation: {
        navigate: jest.fn(),
    },
    id: 'Asteroid Dashboard'
}

const feature = loadFeature('./app/screens/AsteroidDashboard/__tests__/features/AsteroidDashboard.scenarios.feature');

defineFeature(feature, (test) => {
    test('User navigates Asteroid Dashboard', ({ given, when, then }) => {
        let AsteroidDashboardWrapper: ShallowWrapper;
        let AsteroidDashboardInstance: AsteroidDashboard;

        given('I am user loading Asteroid Dashboard Screen', () => {
            AsteroidDashboardWrapper = shallow(<AsteroidDashboard {...screenProps} />);
        });

        when('I just navigate to the Asteroid Dashboard Screen', () => {
            AsteroidDashboardInstance = AsteroidDashboardWrapper.instance() as AsteroidDashboard;
        });

        then('Asteroid Dashboard will mount with out any errors', () => {
            expect(AsteroidDashboardWrapper).toBeTruthy();
            expect(AsteroidDashboardWrapper).toMatchSnapshot();
        });

        then('I can now enter asteroid id', () => {
            let textInputComponent = AsteroidDashboardWrapper.findWhere((node) => node.prop('testID') === 'searchTextInputID');
            textInputComponent.simulate('changeText', '2000887');
            expect(AsteroidDashboardWrapper).toMatchSnapshot();
        });

        then('I can press the asteroid submit button to fetch the asteroid data', () => {
            let submitButtonComponent = AsteroidDashboardWrapper.findWhere((node) => node.prop('testID') === 'submitButtonTestID');
            submitButtonComponent.simulate('press');
            expect(AsteroidDashboardWrapper).toMatchSnapshot();
        });

        then('I can press the asteroid random button to find the random asteroid list', () => {
            let randomButtonComponent = AsteroidDashboardWrapper.findWhere((node) => node.prop('testID') === 'randomButtonTestID');
            randomButtonComponent.simulate('press');
            expect(AsteroidDashboardWrapper).toMatchSnapshot();
        });

        then('I can unmount the screen with out any errors', () => {
            AsteroidDashboardInstance.componentWillUnmount();
            expect(AsteroidDashboardWrapper).toBeTruthy();
            expect(AsteroidDashboardWrapper).toMatchSnapshot();
        });
    })
})