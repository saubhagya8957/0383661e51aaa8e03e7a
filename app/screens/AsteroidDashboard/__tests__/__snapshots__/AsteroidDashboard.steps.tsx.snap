// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asteroid Dashboard Screen Find Asteroid Data User navigates Asteroid Dashboard 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ab8e00",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <KeyboardAwareScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardOpeningTime={250}
    viewIsInsideTabBar={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ab8e00",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#e8f4ff",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 50,
            "textAlign": "center",
          }
        }
      >
        Search Asteroid Data
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Enter Asteroid ID"
        placeholderTextColor="#ffffff"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#000000",
            "borderWidth": 2,
            "color": "#000000",
            "height": 50,
            "marginTop": 20,
            "paddingLeft": 20,
            "width": 300,
          }
        }
        testID="searchTextInputID"
        underlineColorAndroid="transparent"
        value=""
      />
      <ForwardRef
        disabled={true}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderRadius": 15,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 20,
              "width": 250,
            },
            Object {
              "backgroundColor": "#757575",
            },
          ]
        }
        testID="submitButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Submit
        </Text>
      </ForwardRef>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3000a8",
            "borderRadius": 15,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "width": 250,
          }
        }
        testID="randomButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Random Asteroid
        </Text>
      </ForwardRef>
    </View>
  </KeyboardAwareScrollView>
</ForwardRef(SafeAreaView)>
`;

exports[`Asteroid Dashboard Screen Find Asteroid Data User navigates Asteroid Dashboard 2`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ab8e00",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <KeyboardAwareScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardOpeningTime={250}
    viewIsInsideTabBar={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ab8e00",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#e8f4ff",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 50,
            "textAlign": "center",
          }
        }
      >
        Search Asteroid Data
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Enter Asteroid ID"
        placeholderTextColor="#ffffff"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#000000",
            "borderWidth": 2,
            "color": "#000000",
            "height": 50,
            "marginTop": 20,
            "paddingLeft": 20,
            "width": 300,
          }
        }
        testID="searchTextInputID"
        underlineColorAndroid="transparent"
        value="2000887"
      />
      <ForwardRef
        disabled={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderRadius": 15,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 20,
              "width": 250,
            },
            Object {
              "backgroundColor": "#00c92f",
            },
          ]
        }
        testID="submitButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Submit
        </Text>
      </ForwardRef>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3000a8",
            "borderRadius": 15,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "width": 250,
          }
        }
        testID="randomButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Random Asteroid
        </Text>
      </ForwardRef>
    </View>
  </KeyboardAwareScrollView>
</ForwardRef(SafeAreaView)>
`;

exports[`Asteroid Dashboard Screen Find Asteroid Data User navigates Asteroid Dashboard 3`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ab8e00",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <KeyboardAwareScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardOpeningTime={250}
    viewIsInsideTabBar={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ab8e00",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#e8f4ff",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 50,
            "textAlign": "center",
          }
        }
      >
        Search Asteroid Data
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Enter Asteroid ID"
        placeholderTextColor="#ffffff"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#000000",
            "borderWidth": 2,
            "color": "#000000",
            "height": 50,
            "marginTop": 20,
            "paddingLeft": 20,
            "width": 300,
          }
        }
        testID="searchTextInputID"
        underlineColorAndroid="transparent"
        value="2000887"
      />
      <ForwardRef
        disabled={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderRadius": 15,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 20,
              "width": 250,
            },
            Object {
              "backgroundColor": "#00c92f",
            },
          ]
        }
        testID="submitButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Submit
        </Text>
      </ForwardRef>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3000a8",
            "borderRadius": 15,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "width": 250,
          }
        }
        testID="randomButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Random Asteroid
        </Text>
      </ForwardRef>
    </View>
  </KeyboardAwareScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "position": "absolute",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#ffffff"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</ForwardRef(SafeAreaView)>
`;

exports[`Asteroid Dashboard Screen Find Asteroid Data User navigates Asteroid Dashboard 4`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ab8e00",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <KeyboardAwareScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardOpeningTime={250}
    viewIsInsideTabBar={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ab8e00",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#e8f4ff",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 50,
            "textAlign": "center",
          }
        }
      >
        Search Asteroid Data
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Enter Asteroid ID"
        placeholderTextColor="#ffffff"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#000000",
            "borderWidth": 2,
            "color": "#000000",
            "height": 50,
            "marginTop": 20,
            "paddingLeft": 20,
            "width": 300,
          }
        }
        testID="searchTextInputID"
        underlineColorAndroid="transparent"
        value="2000887"
      />
      <ForwardRef
        disabled={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderRadius": 15,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 20,
              "width": 250,
            },
            Object {
              "backgroundColor": "#00c92f",
            },
          ]
        }
        testID="submitButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Submit
        </Text>
      </ForwardRef>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3000a8",
            "borderRadius": 15,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "width": 250,
          }
        }
        testID="randomButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Random Asteroid
        </Text>
      </ForwardRef>
    </View>
  </KeyboardAwareScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "position": "absolute",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#ffffff"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</ForwardRef(SafeAreaView)>
`;

exports[`Asteroid Dashboard Screen Find Asteroid Data User navigates Asteroid Dashboard 5`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ab8e00",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <KeyboardAwareScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardOpeningTime={250}
    viewIsInsideTabBar={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ab8e00",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#e8f4ff",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 50,
            "textAlign": "center",
          }
        }
      >
        Search Asteroid Data
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Enter Asteroid ID"
        placeholderTextColor="#ffffff"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#000000",
            "borderWidth": 2,
            "color": "#000000",
            "height": 50,
            "marginTop": 20,
            "paddingLeft": 20,
            "width": 300,
          }
        }
        testID="searchTextInputID"
        underlineColorAndroid="transparent"
        value="2000887"
      />
      <ForwardRef
        disabled={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderRadius": 15,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 20,
              "width": 250,
            },
            Object {
              "backgroundColor": "#00c92f",
            },
          ]
        }
        testID="submitButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Submit
        </Text>
      </ForwardRef>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3000a8",
            "borderRadius": 15,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "width": 250,
          }
        }
        testID="randomButtonTestID"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Random Asteroid
        </Text>
      </ForwardRef>
    </View>
  </KeyboardAwareScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "position": "absolute",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#ffffff"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</ForwardRef(SafeAreaView)>
`;
